chemprop/cli/predict.py

```diff
@@ -183,110 +227,131 @@ def make_prediction_for_models(
+
+    if args.evaluation_methods is not None:
+        uncertinaty_evaluators = [
+            Factory.build(UncertaintyEvaluatorRegistry[method])
+            for method in args.evaluation_methods
+        ]
+        logger.info("Uncertainty evaluation metric:")
+        for evaluator in uncertinaty_evaluators:
+            test_targets = test_loader.dataset.Y
+            test_mask = torch.from_numpy(np.isfinite(test_targets))
+            test_targets = np.nan_to_num(test_targets, nan=0.0)
+            test_targets = torch.from_numpy(test_targets)
+            if isinstance(evaluator, RegressionEvaluator):
+                metric_value = evaluator.evaluate(test_preds, test_uncs, test_targets, test_mask)
+            else:
+                metric_value = evaluator.evaluate(test_uncs, test_targets, test_mask)
+            logger.info(f"{evaluator.alias}: {metric_value.tolist()}")
+
+    if args.uncertainty_method == "none" and (
+        isinstance(model.predictor, MveFFN) or isinstance(model.predictor, EvidentialFFN)
+    ):
+        test_preds = test_preds[..., 0]
+        test_individual_preds = test_individual_preds[..., 0]
 
-    output_columns = load_output_columns(model_paths[0])
     if output_columns is None:
         output_columns = [
-            f"pred_{i}" for i in range(average_preds.shape[1])
+            f"pred_{i}" for i in range(test_preds.shape[1])
         ]  # TODO: need to improve this for cases like multi-task MVE and multi-task multiclass
 
+    save_predictions(args, model, output_columns, test_preds, test_uncs, output_path)
+
+    if len(model_paths) > 1:
+        save_individual_predictions(
+            args,
+            model,
+            model_paths,
+            output_columns,
+            test_individual_preds,
+            test_individual_uncs,
+            output_path,
+        )
+
+
+def save_predictions(args, model, output_columns, test_preds, test_uncs, output_path):
     if isinstance(model.predictor, MulticlassClassificationFFN):
         output_columns = output_columns + [f"{col}_prob" for col in output_columns]
-        predicted_class_labels = average_preds.argmax(axis=-1)
+        predicted_class_labels = test_preds.argmax(axis=-1)
         formatted_probability_strings = np.apply_along_axis(
-            lambda x: ",".join(map(str, x)), 2, average_preds
+            lambda x: ",".join(map(str, x)), 2, test_preds
         )
-        average_preds = np.concatenate(
+        test_preds = np.concatenate(
             (predicted_class_labels, formatted_probability_strings), axis=-1
         )
 
     df_test = pd.read_csv(
         args.test_path, header=None if args.no_header_row else "infer", index_col=False
     )
-    df_test[output_columns] = average_preds
+    df_test[output_columns] = test_preds
+
+    if args.uncertainty_method not in ["none", "classification"]:
+        unc_columns = [f"{col}_unc" for col in output_columns]
+        df_test[unc_columns] = test_uncs
```

Comment:
I suggest printing less decimal places than is the default. Maybe 6 at most. Right now it gives numbers like `1.7383126678770482`.

Reply:
Good idea